src/lib.rs

```diff
@@ -13,6 +13,103 @@
    See the License for the specific language governing permissions and
    limitations under the License.
 */
+//! ```rust,no_run
+//! // This example connects a `StackDriverExporter` to the `tracing` ecosystem. There's a lot of
+//! // pieces and API layers here, so this example is broken into _many_ steps.
+//!
+//! // The path to your Google Cloud credentials to use with this crate.
+//! let credentials_path = "my_creds.json";
+//!
+//! // The path you want to persist an OAuth2 token. Generally, the lifetime of an OAuth2 token is
+//! // about 1 hour.
+//! let persistent_token_file = Some("oauth_token.json");
+//!
+//! // The amount of time that you, the client, want to give to in-flight span export requests to
+//! // finish. If you specify `None`, then this defaults to 5 seconds.
+//! let maximum_shutdown_duration = Some(Duration::from_secs(10));
+//!
+//! // The number of span export requests that are allowed to be in-flight at once. If you specify
+//! // `None`, it defaults to 0, meaning, no limit will be enforced.
+//! let num_concurrent_requests = 2;
+//!
+//! // An implementation of `futures::task::Spawn`, which allows .
+//! let spawn = {
+//!   // Neither Tokio nor `async-std` provide APIs implementing `futures::task::Spawn`, but
+//!   // `opentelemetry-stackdriver` uses it because it's a useful abstraction over executor
+//!   // runtimes.
+//!   //
+//!   // https://github.com/tokio-rs/tokio/issues/2018
+//!   // https://github.com/async-rs/async-std/issues/142
+//!   pub struct TokioSpawner;
+//!
+//!   impl futures::task::Spawn for TokioSpawner {
+//!     fn spawn_obj(
+//!         &self,
+//!         future: futures::future::FutureObj<'static, ()>
+//!     ) -> Result<(), futures::task::SpawnError> {
+//!       // TODO: check that executor is active; return SpawnError if not.
+//!       tokio::runtime::Handle::current().spawn(future);
+//!       Ok(())
+//!     }
+//!   }
+//! };
+//!
+//! // `StackDriverExporter`: the focal point of this API.
+//! //
+//! // Now that we have all of our configuration pieces, we can construct one!
+//! let exporter = StackDriverExporter::connect(
```

Comment:
It looks like `connect()` currently takes 5 arguments rather than 4, so this isn't actually getting doctested?

Reply:
Fixed the specific issue raised here with 6345ad29 (and decided to add 213ab2e6, in case it's interesting).

Nope. I suppose I could quickly test the code by using another `example/`...hmm.

Reply:
Yeah, there were a bunch of other compile errors, which have been fixed with 4f890ec9. Glad you mentioned this, because that's a big hole in this...err...whole...contribution. 🙂

For reference, I tested the entirety of this PR by adding another `examples/docs.rs` entry and wrapping it with:

```rust
#[tokio::main]
async fn main() {
   // snip
}
```